sklearn/tests/test_pipeline.py

```diff
@@ -868,9 +868,10 @@ def test_pipeline_wrong_memory():
     memory = 1
     cached_pipe = Pipeline([('transf', DummyTransf()), ('svc', SVC())],
                            memory=memory)
-    assert_raises_regex(ValueError, "'memory' should either be a string or a"
-                        " sklearn.externals.joblib.Memory instance, got",
-                        cached_pipe.fit, X, y)
+    assert_raises_regex(ValueError, "'memory' is not a string or a "
+                        "Memory instance implementing a cache method."
+                        " Got a {} instance, instead."
+                        .format(type(memory)), cached_pipe.fit, X, y)
```

Comment:
Still do not make a `type` checking. Apply the same than what you did in the previous commit.